### PR TITLE
Fix `declaration-property-value-no-unknown` reported ranges and message

### PR DESCRIPTION
**Pull request by romainmenke**
> Which issue, if any, is this issue related to?

Closes https://github.com/stylelint/stylelint/issues/8002

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

File: lib/rules/declaration-property-value-no-unknown/__tests__/index.mjs [hunk 1]
```diff
@@ -150,6 +150,12 @@ testRule({
 				}
 			`,
 		},
+		{
+			code: stripIndent`a {
+				top: /* stylelint-disable-next-line declaration-property-value-no-unknown */
+					red;
+			}`,
```
Comment (romainmenke) [inline lib/rules/declaration-property-value-no-unknown/__tests__/index.mjs:157]: A bit of extra test coverage in light of https://github.com/stylelint/stylelint/issues/7904

File: lib/rules/declaration-property-value-no-unknown/__tests__/index.mjs [hunk 2]
```diff
@@ -249,13 +255,24 @@ testRule({
...
+		},
 		{
 			code: 'a { color: ; }',
 			message: messages.rejected('color', ''),
 			line: 1,
-			column: 11,
+			column: 12,
 			endLine: 1,
-			endColumn: 12,
+			endColumn: 13,
```
Comment (romainmenke) [inline lib/rules/declaration-property-value-no-unknown/__tests__/index.mjs:275]: Spaces are valid.
It is the unexpected early `;` that is invalid.

Comment (Mouvedia) [reply lib/rules/declaration-property-value-no-unknown/__tests__/index.mjs:275]: l202–209 is a variant without the space

File: lib/rules/declaration-property-value-no-unknown/index.mjs [hunk 3]
```diff
@@ -142,7 +144,7 @@ const rule = (primary, secondaryOptions) => {
 			let cssTreeValueNode;
 
 			try {
-				cssTreeValueNode = parse(value, { context: 'value' });
+				cssTreeValueNode = parse(value, { context: 'value', positions: true });
```
Comment (Mouvedia) [inline lib/rules/declaration-property-value-no-unknown/index.mjs:147]: https://github.com/csstree/csstree/blob/master/docs/parsing.md#positions

**Pull request merged**